sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala

```diff
@@ -111,7 +111,7 @@ trait GpuPartitioning extends Partitioning with Arm {
         sliceInternalOnGpu(numRows, partitionIndexes, partitionColumns)
       } else {
         sliceInternalOnCpu(numRows, partitionIndexes, partitionColumns)
-      }
+      }.filter(_ != null)
```

Comment:
is it safe to add this here, I checked some of them like (GpuSinglePartitioning,  GpuRangePartitioner) don't filter out the null? 

BTW, if it is safe, This PR should get rid of the other redundant filter

Reply:
> is it safe to add this here, I checked some of them like (GpuSinglePartitioning, GpuRangePartitioner) don't filter out the null?
> 

That's why I add the filtering here, to make sure it takes effect for all the callers.

Reply:
Removed the redundant filter

Reply:
Some tests failed, yes, it is not safe to do the filtering here. Updated